js/app/controllers/calcontroller.js

```diff
@@ -133,6 +133,38 @@ app.controller('CalController', ['$scope', 'Calendar', 'CalendarService', 'VEven
 			$scope.$apply();
 		});
 
+		$scope.importFileById = function(id) {
```

Comment:
Indeed.
Maybe I'm over-engineering here again, but we could at a HashService that parses the hash and other services could register for certain pattern of hashes.

I definitely see multiple use cases, where we need hashes in the future:
- this pr for importing a file
- linking directly to an event from a different app
- (linking directly to a date (-range) from a different app)
- register a protocol handler for adding webcal subscriptions